control-plane-operator/controllers/hostedcontrolplane/cloud/openstack/providerconfig.go

```diff
@@ -6,43 +6,57 @@ import (
 )
 
 const (
-	CloudConfigKey = "clouds.conf"
-	CaKey          = "ca.pem"
-	Provider       = "openstack"
+	CloudConfigDir      = "/etc/openstack/config"
+	CloudCredentialsDir = "/etc/openstack/credentials"
+	CredentialsFile     = "clouds.conf"
+	CaDir               = "/etc/pki/ca-trust/extracted/pem"
+	CaKey               = "ca.pem"
+	Provider            = "openstack"
+	cloudsSecretKey     = "clouds.yaml"
+	caSecretKey         = "cacert"
 )
 
 // ReconcileCloudConfig reconciles as expected by Nodes Kubelet.
-func ReconcileCloudConfig(secret *corev1.Secret, hcp *hyperv1.HostedControlPlane, credentialsSecret *corev1.Secret) error {
+func ReconcileCloudConfig(secret *corev1.Secret, hcp *hyperv1.HostedControlPlane, credentialsSecret *corev1.Secret, hasCACert bool) error {
 	if secret.Data == nil {
 		secret.Data = map[string][]byte{}
 	}
-	config := string(credentialsSecret.Data[CloudConfigKey]) // TODO(dulek): Missing key handling
+	config := string(credentialsSecret.Data[CredentialsFile]) // TODO(dulek): Missing key handling
 
 	config += `
 [Global]
 use-clouds=true
-clouds-file=/etc/openstack/credentials/clouds.yaml`
+clouds-file=` + CloudCredentialsDir + "/" + CredentialsFile + "\n"
 
 	config += "\ncloud=" + hcp.Spec.Platform.OpenStack.IdentityRef.CloudName
 
 	// FIXME(dulek): This is specific to CCM, we might want to have 2 versions.
 	// FIXME(dulek): Is it really a good idea to have it here?
 	// FIXME(dulek): How do we make it configurable?
-	if hcp.Spec.Platform.OpenStack.CACertSecret != nil {
-		config += "\nca-file = /etc/pki/ca-trust/extracted/pem/ca.pem\n"
+	if hasCACert {
+		config += "\nca-file =" + CaDir + "/" + CaKey + "\n"
 	}
 
 	config += "\n[LoadBalancer]\nmax-shared-lb = 1\nmanage-security-groups = true\n"
 
-	secret.Data[CloudConfigKey] = []byte(config)
+	secret.Data[CredentialsFile] = []byte(config)
 	return nil
 }
 
 // ReconcileTrustedCA reconciles as expected by Nodes Kubelet.
-func ReconcileTrustedCA(cm *corev1.ConfigMap, hcp *hyperv1.HostedControlPlane, caConfigMap *corev1.Secret) error {
+func ReconcileTrustedCA(cm *corev1.ConfigMap, hcp *hyperv1.HostedControlPlane, caCertData []byte) error {
 	if cm.Data == nil {
 		cm.Data = map[string]string{}
 	}
-	cm.Data[CaKey] = string(caConfigMap.Data[CaKey]) // TODO(dulek): Missing key handling
+	cm.Data[CaKey] = string(caCertData) // TODO(dulek): Missing key handling
 	return nil
 }
+
+// GetCloudConfigFromCredentialsSecret returns the CA cert from the credentials secret.
+func GetCACertFromCredentialsSecret(secret *corev1.Secret) []byte {
+	caCert, ok := secret.Data[caSecretKey]
```

Comment:
hmm this is wrong. It's not the CA path but the actual CA content. I need to fix this.